honeycombio/resource_slo.go

```diff
@@ -39,9 +39,31 @@ func newSLO() *schema.Resource {
 			},
 			"dataset": {
 				Type:        schema.TypeString,
-				Required:    true,
+				Optional:    true,
 				ForceNew:    true,
 				Description: "The dataset this SLO is created in. Must be the same dataset as the SLI unless the SLI's dataset is `\"__all__\"`.",
+				DiffSuppressFunc: func(k, oldValue, newValue string, d *schema.ResourceData) bool {
+					if oldValue == newValue {
+						return true
+					}
+					// if the config moves away from deprecated dataset, nothing should change
+					if newValue == "" || newValue == "__all__" {
+						return true
+					}
+					return false
+				},
```

Comment:
If we're going to ship this functionality, I'd like to see a test for it.

We're also not currently setting an MDSLOs dataset to `__all__` (you're asserting the property is empty below.